app/modules/teams/components/helpers/useTeamAuthorization.ts

```diff
@@ -0,0 +1,26 @@
+import { useContext } from 'react';
+import { AuthenticationContext } from '~/modules/authentication/containers/authentication.container';
+import type { User } from '~/modules/users/users.types';
+import TeamAuthorization from '../../authorization';
+
+export default function useTeamAuthorization(teamId: string | null = null) {
+  const user = useContext(AuthenticationContext) as User | null;
+
+  const defaults = {
+    canCreate: user ? TeamAuthorization.canCreate(user) : false,
+    canView: false,
+    canUpdate: false,
+    canDelete: false,
+    canInviteUsers: false,
+  };
+
+  if (!user || !teamId) return defaults;
+
+  return {
+    ...defaults,
+    canView: TeamAuthorization.canView(user, teamId),
```

Comment:
I wonder if you need to be more specific on the exported names incase of a collision or we have permissions for something else?

`canViewTeam`